src/System.IO.Pipelines/tests/PipeWriterStreamTests.nonnetstandard.cs

```diff
@@ -167,6 +167,15 @@ public async Task DefaultPipeWriterImplementationReturnsPipeWriterStream()
             Assert.True(pipeWriter.FlushCalled);
         }
 
+        [Fact]
+        public void AsStreamDoNotCompleteWriter()
+        {
+            var pipeWriter = new TestPipeWriter();
+
+            // will throw in Complete if it was actually invoked
```

Comment:
My concern here is that someone comes along someday and deciedes `TestPipeWriter.Complete` shouldn't throw and then the `leaveOpen` behavior is also broken. In that situation, this test won't fail. This is probably worth adding a `ThrowsOnCompletePipeWriter` (and similar for the pipe reader) just to make sure the behavior is super clear in the test.